optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberAccessorImplementor.java

```diff
@@ -0,0 +1,842 @@
+        ResultHandle annotatedElement = methodCreator.invokeStaticMethod(getAnnotatedElementGetter(),
+                gizmoMemberAccessorNameResult);
+        ResultHandle out = methodCreator.invokeInterfaceMethod(getAnnotationMethod("getDeclaredAnnotations"),
+                annotatedElement);
+        methodCreator.returnValue(out);
+    }
+
+    /**
+     * Describe and provide simplified/unified access for a Member
+     */
+    public static class MemberDescriptor {
+
+        /**
+         * The name of a member. For a field, it the field name.
+         * For a method, if it is a getter, the method name without "get"/"is"
+         * and the first letter lowercase; otherwise, the method name.
+         */
+        String name;
+
+        /**
+         * If the member is a field, the FieldDescriptor of the member
+         * If the member is a method, the MethodDescriptor of the member
+         */
+        Object memberDescriptor;
+
+        /**
+         * If the member is a normal member, the class that declared it
+         * If the member is from Jandex, the Jandex ClassInfo of the class that declared it
+         */
+        Object declaringClass;
+
+        /**
+         * The member as an AnnotatedElement
+         */
+        AnnotatedElement annotatedElement;
+
+        /**
+         * The MethodDescriptor of the corresponding setter. Is empty if not present.
+         */
+        Optional<MethodDescriptor> setter;
+
+        /**
+         * The generic type of the member
+         */
+        Type type;
+
+        public MemberDescriptor(Member member) {
+            declaringClass = member.getDeclaringClass();
+            if (member instanceof Field) {
+                memberDescriptor = FieldDescriptor.of((Field) member);
+                name = member.getName();
+                annotatedElement = (Field) member;
+                type = ((Field) member).getGenericType();
+                setter = lookupSetter();
+            } else if (member instanceof Method) {
+                memberDescriptor = MethodDescriptor.ofMethod((Method) member);
+                annotatedElement = (Method) member;
+                if (ReflectionHelper.isGetterMethod((Method) member)) {
+                    name = ReflectionHelper.getGetterPropertyName(member);
+                } else {
+                    name = member.getName();
+                }
+                type = ((Method) member).getGenericReturnType();
+                setter = lookupSetter();
+            } else {
+                throw new IllegalArgumentException(member + " is not a Method or a Field.");
+            }
+        }
+
+        // For Quarkus
+        // (Cannot move to Quarkus module; get runtime
+        //  exception since objects created here use classes
+        //  from another ClassLoader).
+        public MemberDescriptor(FieldInfo fieldInfo, IndexView indexView) {
```

Comment:
When there are several constructors with various combinations of parameters, it's often better to reuse the most general one that takes all the possible parameters, possibly together with factory methods.